astor/code_gen.py

```diff
@@ -53,7 +53,7 @@ class that will be instantiated and used to generate the source code.
     """
     if source_generator_class is None:
         source_generator_class = SourceGenerator
-    elif not isinstance(source_generator_class, SourceGenerator):
+    elif not issubclass(source_generator_class, SourceGenerator):
         raise TypeError('source_generator_class should be a subclass of SourceGenerator')
     elif not callable(source_generator_class):
```

Comment:
I think this check can be removed now.

Reply:
I ended up moving the check to `inspect.isclass` to provide a more interpretable error if the user passes in any kind of object.